core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -71,13 +71,8 @@ public DeterministicKey(List<ChildNumber> childNumberPath,
                             LazyECPoint publicAsPoint,
                             @Nullable BigInteger priv,
                             @Nullable DeterministicKey parent) {
-        super(priv, publicAsPoint.compress());
-        checkArgument(chainCode.length == 32);
-        this.parent = parent;
-        this.childNumberPath = HDPath.M(Objects.requireNonNull(childNumberPath));
-        this.chainCode = Arrays.copyOf(chainCode, chainCode.length);
-        this.depth = parent == null ? 0 : parent.depth + 1;
-        this.parentFingerprint = (parent != null) ? parent.getFingerprint() : 0;
+        this(priv, publicAsPoint.compress(), parent == null ? 0 : parent.depth + 1, parent,
+                parent != null ? parent.getFingerprint() : 0, chainCode, HDPath.M(childNumberPath), null, null);
```

Comment:
Since `priv` is nullable, `makeHDPath` 7bc87972523443ab910146b6622eb6acce767ae7 could be used here. Probably in a follow-on commit.